File: frontend/src/api/hackathonService.js
import api from './axios';

export const hackathonAPI = {
  getAll: () => api.get('/hackathons'),
  getById: (id) => api.get(`/hackathons/${id}`),
  getByStatus: (status) => api.get(`/hackathons/status/${status}`),
  create: (data) => api.post('/hackathons', data),
  update: (id, data) => api.put(`/hackathons/${id}`, data),
  delete: (id) => api.delete(`/hackathons/${id}`),
};

File: backend/src/main/java/com/hackathon/service/HackathonService.java
package com.hackathon.service;

import com.hackathon.dto.request.HackathonRequest;
import com.hackathon.model.Hackathon;

import java.util.List;

public interface HackathonService {

    Hackathon create(HackathonRequest request, String userId);

    Hackathon update(String id, HackathonRequest request);

    void delete(String id);

    Hackathon getById(String id);

    List<Hackathon> getAll();

    List<Hackathon> getByStatus(String status);

    com.hackathon.dto.response.HackathonStatsResponse getHackathonStats(String hackathonId);
}
